util/health/health.go

```diff
@@ -59,17 +59,21 @@ func IsWorse(current, new appv1.HealthStatusCode) bool {
 }
 
 func getIngressHealth(obj *unstructured.Unstructured) (*appv1.HealthStatus, error) {
+	health := appv1.HealthStatus{Status: appv1.HealthStatusProgressing}
 	obj, err := kube.ConvertToVersion(obj, "extensions", "v1beta1")
 	if err != nil {
-		return nil, err
+		health.Status = appv1.HealthStatusUnknown
+		health.StatusDetails = err.Error()
+		return &health, err
```

Comment:
It looks like we are repeating this logic over and over again. It's better to move the setting of the Status and StatusDetails to the caller of all of these functions if it returns an error.